### DISCUSSION TITLE
Unable to change image generation settings or turn feature on. (crashes session)

### DISCUSSION THREAD

**Discussion by azreeel** [open-webui/open-webui#6707] [Issues]
## Installation Method

Docker

## Environment

- Open WebUI Version: v0.3.35
- Ollama Version: 0.3.14

- Operating System: MX Linux 23.4 (Debian 12.7)
- Browser: Replicated in Firefox, Chrome, Opera GX

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Able to change settings and test connection in the Images portion of admin settings

## Actual Behavior:

Browser session will crash upon changing any settings or testing connection (within admin:Images)
After the crash, no button navigation on the entire page will work. Fixed by refreshing page.
All settings are reverted to before attempting to change anything.

## Description

**Bug Summary:**
I've had this issue with the image admin settings for a little while. All other parts of the app work perfectly, but I can no longer change any settings in the image settings and cannot enable the feature. Issue persisted after updating ollama, updating open-webui, removing and recreating docker container.

## Reproduction Details

**Steps to Reproduce:**
Started happening around the first time I updated open-webui to v0.3.32 using watchtower. Unsure if it is related.

To replicate issue, I just need to test connection on the images admin settings. Issue also can be triggered by changing image generation engine or attempting to turn on the image generation feature.

## Logs and Screenshots

**Browser Console Logs:**
POST https://*myurl*/images/api/v1/config/update 422 (Unprocessable Content) index.ts:38
Uncaught (in promise) TypeError: t is not a constructor  dom.js:1228
    at se (dom.js:1228:9)
    at Array.Xi (Toast.svelte:338:32)
    at dn (Toast.svelte:335:42)
    at Array.Vi (Toast.svelte:330:20)
    at Gi (Toast.svelte:304:22)
    at Pt (Component.js:148:34)
    at new no (Toast.svelte:179:5)
    at An (Toaster.svelte:216:40)
    at Sn (Toaster.svelte:194:11)
    at Bn (Toaster.svelte:163:9)      
GET https://*myurl*/images/api/v1/config/url/verify 400 (Bad Request) index.ts:73 

**Docker Container Logs:**
INFO:     10.1.1.1:60777 - "GET /images/api/v1/config HTTP/1.1" 200 OK
INFO:     10.1.1.1:33739 - "GET /images/api/v1/image/config HTTP/1.1" 200 OK
INFO:     10.1.1.1:60777 - "POST /images/api/v1/config/update HTTP/1.1" 422 Unprocessable Entity
INFO  [open_webui.env] Saving 'ENABLE_IMAGE_GENERATION' to the database
INFO:     10.1.1.1:33739 - "GET /images/api/v1/config/url/verify HTTP/1.1" 400 Bad Request
INFO:     10.1.1.1:60777 - "POST /images/api/v1/config/update HTTP/1.1" 422 Unprocessable Entity
INFO:     10.1.1.1:33739 - "GET /api/config HTTP/1.1" 200 OK
INFO  [open_webui.env] Saving 'ENABLE_IMAGE_GENERATION' to the database
INFO:     10.1.1.1:60777 - "GET /images/api/v1/models HTTP/1.1" 400 Bad Request
INFO:     10.1.1.1:33739 - "POST /images/api/v1/config/update HTTP/1.1" 422 Unprocessable Entity
INFO:     10.1.1.1:60777 - "POST /images/api/v1/config/update HTTP/1.1" 422 Unprocessable Entity
INFO:     10.1.1.1:33739 - "GET /api/config HTTP/1.1" 200 OK
INFO  [open_webui.env] Saving 'ENABLE_IMAGE_GENERATION' to the database
INFO:     10.1.1.1:60777 - "GET /images/api/v1/models HTTP/1.1" 400 Bad Request

## Additional Information

When I first tried updating open-webui, I used the persistent watchtower container using the instructions here: https://docs.openwebui.com/getting-started/updating/
It never updated my open-webui container, so I removed the persistent container and did the run once watchtower method. That method actually updated open-webui. It was around this time that I noticed the image settings issue, but cannot 100% say it broke from the update. I have updated since then and have tried removing and recreating the docker container but the issue still persists.

**Comment by ioddevs** [comment]
same.... :(
i am on docker...

i am not sure where to cleanup this settings seems broken somewhere

**Reply by azreeel** [reply]
I was able to fix it by wiping out the docker volume it uses to store data. But, thats not an optimal solution as it loses all your chats, settings, users, etc.
Need to find a way to only wipe out the image settings. (And the project probably needs an improvement to not break entirely after bad requests)

**Reply by azreeel** [reply]
Managed to break it again in the same way by calling a generate image function while the stable diffuison api was turned off.

**Reply by turbra** [reply]
I was able to remove the image settings from the database by executing the following:
```
UPDATE config
SET data = json_remove(data, '$.image_generation')
WHERE id = 1;
```

**Reply by RogerSik** [reply]
Thanks @turbra 

For Postgres User like me:
```
UPDATE config
SET data = data::jsonb - 'image_generation'
WHERE id = 1;
```

I needed to restart open-webui otherwise open-webui showed still the old (deleted) data.
